### PR TITLE
Issue #3566: adding and turning off omitIgnoredModules

### PR DESCRIPTION
**Pull request by rnveach**
Issue #3566

### CODE REVIEW

**Comment by rnveach** [comment]
> Numeric toString() called in an internationalized context

IntelliJ error seems like a false positive. Boolean is not a numeric and it shouldn't cause an issue, no matter the internationalization, as we pass it right back to `Boolean.parse`.
Should I suppress or just change the `Boolean.TRUE.toString()` to `"true"`?

**Comment by romani** [comment]
Please make it `"true"` , making mess with ignore annotations of IntellijIdea does not make sense there.  Refactored code will be the same and may be a bit clear. if value is changed you will write the same characters , difference is only in lower/upper case.

**Comment by rnveach** [comment]
Done.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java [hunk 1]
```diff
@@ -70,6 +70,7 @@
         + " -j,--javadocTree            Print Parse tree of the Javadoc comment%n"
         + " -J,--treeWithJavadoc        Print full Abstract Syntax Tree of the file%n"
         + " -o <arg>                    Sets the output file. Defaults to stdout%n"
+        + " -omitIgnoredModules <arg>   Sets ability to not run modules that are ignored.%n"
```
Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: Does the option have agrs? Do we need ```<arg>```. Should the option have a short name?

Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: It has an argument. `true`/`false`.
I couldn't think of a short name. `o` is already taken and I'm not sure `i` would fit. I questioned adding `oim`.

**Comment by rnveach** [comment]
I also forgot to update CLI documentation. I will do so once option is solidified.

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java (hunk 1 above)
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: It is ok to skip short name for this option, it will not be frequently used.

Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: @MEZk ping

Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: @rnveach 
> If we want to change this to no argument, what should it's name be? We are omitting modules by default.

1. executeIgnoredModules, short: eim
2. runIgnoredModules, short: rim
3. processIgnoredModules, short: pim

In general boolean options should not have arguments in CLI.

Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: Do we need short form for this ?
Short forms exists only to simplify typing for very frequent arguments. I doubt that this option will be used by users, it useful only for our testing tool.

Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: I am leaning towards `executeIgnoredModules`. @romani Are you ok with this?

I don't see a reason for short form right now.

Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: I am ok with name, I like positive names more.

Comment (MEZk) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: Lets drop the idea of having short name for the option. I'm ok.

**Comment by romani** [comment]
Ping

File: src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java (hunk 1 above)
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/MainTest.java]: Done.

**Comment by rnveach** [comment]
CLI documentation was updated.

File: src/main/java/com/puppycrawl/tools/checkstyle/Main.java [hunk 2]
```diff
@@ -405,7 +409,8 @@ private static int runCheckstyle(CliOptions cliOptions)
 
         // create a configuration
         final Configuration config = ConfigurationLoader.loadConfiguration(
-                cliOptions.configLocation, new PropertiesExpander(props));
+                cliOptions.configLocation, new PropertiesExpander(props),
+                cliOptions.omitIgnoredModules);
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/Main.java]: should be executeIgnoredModules

File: src/main/java/com/puppycrawl/tools/checkstyle/Main.java [hunk 3]
```diff
@@ -642,5 +649,7 @@ private static Options buildOptions() {
         private String outputLocation;
         /** List of file to validate. */
         private List<File> files;
+        /** Switch whether to execute ignored modules or not. */
+        private boolean omitIgnoredModules;
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/Main.java]: lets be consistent - `executeIgnoredModules`

File: src/main/java/com/puppycrawl/tools/checkstyle/Main.java [hunk 4]
```diff
@@ -378,6 +381,7 @@ private static CliOptions convertCliToPojo(CommandLine cmdLine, List<File> files
         conf.configLocation = cmdLine.getOptionValue(OPTION_C_NAME);
         conf.propertiesLocation = cmdLine.getOptionValue(OPTION_P_NAME);
         conf.files = filesToProcess;
+        conf.omitIgnoredModules = !cmdLine.hasOption(OPTION_EXECUTE_IGNORED_MODULES_NAME);
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/Main.java]: executeIgnoredModules

**Comment by MEZk** [comment]
@romani 
> If you want you can make old property as deprecated, will be removed in checkstyle8.

Lets deprecate as we did in https://github.com/checkstyle/checkstyle/issues/3671 the property and add the new one.

**Comment by romani** [comment]
@rnveach , to resolve CIs, please rebase.

File: src/main/java/com/puppycrawl/tools/checkstyle/Main.java (hunk 4 above)
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/Main.java]: Done.

File: src/main/java/com/puppycrawl/tools/checkstyle/Main.java (hunk 2 above)
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/Main.java]: Done.

File: src/main/java/com/puppycrawl/tools/checkstyle/Main.java (hunk 3 above)
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/Main.java]: Done.

**Comment by rnveach** [comment]
@romani rebased.
I just renamed ant's property since this will most likely be merged in CS8.
If you still want to deprecate and old and not remove it, let me know.

**Comment by romani** [comment]
@rnveach , please rebase to merge.

**Comment by rnveach** [comment]
Rebased.

**Comment by MEZk** [comment]
@romani @rnveach 
Changes are OK

**Pull request merged**